source-control/git-guidance/readme.md

```diff
@@ -1,11 +1,200 @@
+
+## Cloning
+
+Cloning a repository pulls down a full copy of all the repository data, so that you can work on it locally.
+This copy includes all versions of every file and folder for the project.
+
+```cmd
+git clone https://github.com/username/repo-name
+```
+
+You only need to clone the repository the first time. Before any subsequent branches you can sync any changes from the remote repository using `git pull`.
+
+## Branching
+
+Often, the `main` or `develop` branch of a repository will be locked so that you can't make changes without a Pull Request.
+In this case it is useful to create a separate branch for your local/feature work, so that you can work and track your changes without modifying the main trunk.
+
+Pull the latest changes and create a new branch for your work based on the trunk (in this case develop)
+
+```cmd
+git pull
+git checkout -b feature/feature-name develop
+```
+
+At any point, you can move between the branches with `git checkout <branch>` as long as you have committed or stashed your work.
+If you forget the name of your branch use `git branch --all` to list all branches
+
+When you are done working, push your changes to a branch in the remote repository using `git push`
+
+```cmd
+git push --set-upstream origin feature/feature-name
+```
+
+after the first push, the --set-upstream parameter is not needed
+
+## Committing
+
+To avoid loosing work, it is good to commit often in small chunks. This allows you to revert only the last changes if you discover a problem and also neatly explains exactly what changes were made and why.
+
+1. Make changes to your branch
+
+2. Check what files were changed
+
+    ```cmd
+    > git status
+    On branch feature/271-basic-commit-info
+    Changes not staged for commit:
+      (use "git add <file>..." to update what will be committed)
+      (use "git restore <file>..." to discard changes in working directory)
+            modified:   source-control/git-guidance/readme.md
+    ```
+
+3. Track the files you wish to include in the commit
+
+    ```cmd
+    git add --all
+    ```
+
+   to track all modified files, or
+
+    ```cmd
+    git add source-control/git-guidance/readme.md
+    ```
+
+   to track only specific files
+
+4. Commit the changes to your local branch with a descriptive [commit message](../readme.md#commit-best-practices)
+
+    ```cmd
+    git commit -m "add basic git instructions"
+    ```
+
+## Merging
+
+In [CSE](..\..\CSE.md) we encourage the use of Pull Request to merge code to the main repository to make sure that all code in the final product is [code reviewed](..\..\code-reviews\readme.md)
+
+The Pull Request (PR) process in [Azure DevOps](https://docs.microsoft.com/en-us/azure/devops/repos/git/pull-requests?view=azure-devops), [GitHub](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request) and other similar tools make it easy both to start a PR, review a PR and merge a PR.
```

Comment:
Might be nice to add a line about how you can link issues and ADO board items using #<number> for git issues or #AB<number> for ADO board items. This links them so when the PR is closed, they automatically move to closed.

Reply:
This is already done in the code reviews section so I think I might leave it out of the git guidance